crates/hyperswitch_connectors/src/connectors/novalnet/transformers.rs

```diff
@@ -56,66 +127,191 @@ impl TryFrom<&NovalnetRouterData<&PaymentsAuthorizeRouterData>> for NovalnetPaym
             PaymentMethodData::Card(req_card) => {
-                let card = NovalnetCard {
-                    number: req_card.card_number,
-                    expiry_month: req_card.card_exp_month,
-                    expiry_year: req_card.card_exp_year,
-                    cvc: req_card.card_cvc,
-                    complete: item.router_data.request.is_auto_capture()?,
+                let auth = NovalnetAuthType::try_from(&item.router_data.connector_auth_type)?;
+
+                let merchant = NovalnetPaymentsRequestMerchant {
+                    signature: auth.product_activation_key,
+                    tariff: auth.tariff_id,
+                };
+
+                let novalnet_card = NovalNetPaymentData::PaymentCard(NovalNetCard {
+                    card_number: req_card.card_number,
+                    card_expiry_month: req_card.card_exp_month,
+                    card_expiry_year: req_card.card_exp_year,
+                    card_cvc: Some(req_card.card_cvc),
+                    card_holder: item.router_data.get_optional_billing_full_name(),
+                });
+
+                let enforce_3d = match item.router_data.auth_type {
+                    enums::AuthenticationType::ThreeDs => Some(1),
+                    enums::AuthenticationType::NoThreeDs => None,
+                };
+                let test_mode = match item.router_data.test_mode {
+                    Some(true) => 1,
+                    Some(false) | None => 0,
+                };
+
+                let return_url = item.router_data.request.get_return_url().ok();
+                let hook_url = item.router_data.request.get_webhook_url().ok();
+                let transaction = NovalnetPaymentsRequestTransaction {
+                    test_mode,
+                    payment_type: NovalNetPaymentTypes::CREDITCARD,
+                    amount: Some(item.amount.clone()),
+                    currency: Some(item.router_data.request.currency.to_string()),
+                    order_no: Some(item.router_data.connector_request_reference_id.clone()),
+                    hook_url,
+                    return_url: return_url.clone(),
+                    error_return_url: return_url.clone(),
+                    payment_data: novalnet_card,
+                    enforce_3d,
                 };
+
+                let billing = NovalnetPaymentsRequestBilling {
+                    house_no: item.router_data.get_optional_billing_line1(),
+                    street: item.router_data.get_optional_billing_line2(),
+                    city: item.router_data.get_optional_billing_city(),
+                    zip: item.router_data.get_optional_billing_zip(),
+                    country_code: item.router_data.get_optional_billing_country(),
+                };
+
+                let customer_ip = item
+                    .router_data
+                    .request
+                    .get_browser_info()?
+                    .get_ip_address()?;
```

Comment:
You need to add such required fields in `/crates/router/src/configs/defaults.rs`

Reply:
added